src/agent/configuration_parser/CMakeLists.txt

```diff
@@ -11,7 +11,7 @@ set_common_settings()
 find_package(toml11 CONFIG REQUIRED)
 
 add_library(ConfigurationParser src/configuration_parser.cpp)
-target_include_directories(ConfigurationParser PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(ConfigurationParser PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
```

Comment:
Same, it seems that there's no reason to add the `src` folder as a target include.